### DISCUSSION TITLE
xDrip+ not updating on Galaxy Watch6

### DISCUSSION THREAD

**Discussion by StevenUK1974** [NightscoutFoundation/xDrip#3323] [Q&A]
Hi,

I updated xDrip+ on my phone this morning to the latest Alpha build (2024.01.16) but it hasn't been able to update the version on my Galaxy Watch6 running WearOS 4.0.

Both ADB and Wireless debugging are turned on but even when I try a Force Update on the watch it checks for ADB and then says it's not working without asking me for permission. 

Please can someone help me to get this working? 

Thanks 
Steven 
![Screenshot_20240129_081936_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/158083537/d7c3a3e4-85c6-44eb-b701-93f743f42d54)
![Screenshot_20240129_174059_settings](https://github.com/NightscoutFoundation/xDrip/assets/158083537/413ce049-2351-4ce8-b263-a80865e54858)
![Screenshot_20240129_174113_settings](https://github.com/NightscoutFoundation/xDrip/assets/158083537/72bb425e-303e-49a1-8288-1e7a4d86bcc7)
![Screenshot_20240129_174048_settings](https://github.com/NightscoutFoundation/xDrip/assets/158083537/1cef6839-1903-4464-8c99-d70799eac403)
![Screenshot_20240129_143004_dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/158083537/f3cc93d9-ce79-4d61-a0ad-c42b2a9bba57)
![Screenshot_20240129_154325_sysui](https://github.com/NightscoutFoundation/xDrip/assets/158083537/2bf59209-a021-4239-9200-0e05de3de00a)

**Comment by StevenUK1974** [comment]
Latest screenshots. It never goes above 0%.

![Screenshot_20240130_094826_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/158083537/763bc09b-0ae6-4ea5-b254-ca96621d8577)
![Screenshot_20240130_094843_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/158083537/0d1c2158-58cf-4ed0-b0a5-acc0b4cc828b)
![Screenshot_20240130_095118_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/158083537/22ff3647-5a74-4f2b-9ce9-a62860540ee0)

![Screenshot_20240130_094751_settings](https://github.com/NightscoutFoundation/xDrip/assets/158083537/b6807bb4-8c29-4f73-b285-4c0d733fb6be)
![Screenshot_20240130_094754_settings](https://github.com/NightscoutFoundation/xDrip/assets/158083537/ac1863e5-3ae1-4361-9557-bfbc751501c6)

**Reply by thederekFawcett** [reply]
If xDrip on your phone recently updated, what I do is use Wear Installer 2 to update/reinstall the new xDrip apk to your watch. I don't know why the Wear OS app doesn't update automatically but this is how I do it. Have got 3 phones, 3 watches, and 4 installs and it's a pain in the butt but it works

**Reply by StevenUK1974** [reply]
Hi, yeah that's what I ended up doing last week. I can't see how it could actually work over Wi-Fi as there's nowhere I can see to input the IP address and port of the watch?

**Reply by thederekFawcett** [reply]
So you got it updated?

**Reply by StevenUK1974** [reply]
Yes I did thanks.
